### DISCUSSION TITLE
ejabberdctl cannot connect to node, always returns exit code 1

### DISCUSSION THREAD

**Discussion by TeknikalDomain** [processone/ejabberd#3687] [Questions] [answered]
- ejabberd version: 20.01-1
- erl version: 10.6.4
- Installed via official Ubuntu package repository

At first I thought this was a systemd configuration issue, where it'd *report* as failed but still start, but it's even weirder.
All `ejabberdctl` commands on a fresh install... just don't work. Exit code 1, no response, no meaningful error. This also includes the `ejabberdctl started` command that systemd is waiting on.

But it's *even weirder*: starting with `ejabberdctl start` has the same effect. I can't even `ejabberdctl stop` to shut it down, I have to manually `kill -TERM` the process. Even if I copy the node name straight out of the `--sname` parameter, doesn't do anything.

To summarize: `ejabberdctl` cannot connect to a running `ejabberd` instance locally, only giving a very unhelpful error message of *nothing,* with nothing in the config that should be denying permissions, nothing in the logs, and nothing to indicate what's going on.

Any ideas? that's a brick wall if I've ever seen one.

**Comment by badlop** [comment]
So, you installed from here https://launchpad.net/ubuntu/+source/ejabberd a version that is more than a year and half old, and just after installation, it fails to start?

Did you check if there is any similar reported error in that package bug tracker?

Or is there any recent version with a reported fix for that?

If that package is broken when just fresh installed, then that package may have a bug worth reporting in that package bug tracker, right?

Maybe you can try other package versions from Ubuntu, recent or older versions?

If all that fails, and find no other solution, you can try the [DEB package provided by ProcessOne](https://www.process-one.net/en/ejabberd/downloads/): it probably doesn't integrate gracefully with your Ubuntu, but at least it should... start

**Comment by TeknikalDomain** [comment]
Oh it'll start perfectly fine. Just that, once started, `ejabberdctl` can't perform any meaningful commands.

I can't find any reported fix because it's hard to search for a fix when everyone's favorite ~~un~~reliable source of knowledge, Google, only points to issues with the node name, and "it starts but can't connect" with no logs and no meaningful error isn't something that's easily grepped for.

**Comment by badlop** [comment, marked as answer]
Downloaded from ejabberd.im, installed in my old Debian system:

```
❯ sudo dpkg -i ejabberd_21.07-0_amd64.deb
S'està seleccionant el paquet ejabberd prèviament no seleccionat.
(S'està llegint la base de dades… hi ha 284629 fitxers i directoris instal·lats actualment.)
S'està preparant per a desempaquetar ejabberd_21.07-0_amd64.deb…
S'està desempaquetant ejabberd (21.07-0)…
S'està configurant ejabberd (21.07-0)…
```

**Start ejabberd interactively**

```
❯ sudo /opt/ejabberd-21.07/bin/ejabberdctl live
...
17:41:23.604 [info] ejabberd 21.07 is started in the node ejabberd@localhost in 3.87s

In another console:

❯ sudo /opt/ejabberd-21.07/bin/ejabberdctl status
The node ejabberd@localhost is started with status: started
ejabberd 21.07 is running in that node

❯ sudo /opt/ejabberd-21.07/bin/ejabberdctl registered_vhosts
atenea

❯ sudo /opt/ejabberd-21.07/bin/ejabberdctl register user1 atenea pass123
User user1@atenea successfully registered

❯ sudo /opt/ejabberd-21.07/bin/ejabberdctl stop

```

**Start ejabberd using ejabberd.init**

```
❯ sudo /opt/ejabberd-21.07/bin/ejabberd.init start
Starting ejabberd...
done.

❯ sudo /opt/ejabberd-21.07/bin/ejabberd.init help  
Usage: /opt/ejabberd-21.07/bin/ejabberd.init {start|stop|restart|force-reload|status}

❯ sudo /opt/ejabberd-21.07/bin/ejabberd.init status
Getting ejabberd status...
The node ejabberd@localhost is started with status: started
ejabberd 21.07 is running in that node

❯ sudo /opt/ejabberd-21.07/bin/ejabberdctl status
The node ejabberd@localhost is started with status: started
ejabberd 21.07 is running in that node

❯ sudo /opt/ejabberd-21.07/bin/ejabberdctl registered_vhosts
atenea

❯ sudo /opt/ejabberd-21.07/bin/ejabberdctl register user2 atenea pass456
User user2@atenea successfully registered

❯ sudo /opt/ejabberd-21.07/bin/ejabberd.init stop
Stopping ejabberd...
done.

```
